### PR TITLE
nixVersions.stable: 2.7.0 -> 2.8.0

### PR DESCRIPTION
**Pull request by drupol**
Update to Nix 2.8.0: https://discourse.nixos.org/t/nix-2-8-0-released/18714

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drupol** [comment]
I would like some assistance on how to update the file `nixos/modules/installer/tools/nix-fallback-paths.nix`.

**Comment by lovesegfault** [comment]
Tagging @edolstra for the fallback path info.

**Comment by Kranzes** [comment]
> I would like some assistance on how to update the file `nixos/modules/installer/tools/nix-fallback-paths.nix`.

The paths are generated and updates from [this}(https://github.com/NixOS/nix/blob/master/maintainers/upload-release.pl) messy and hard-coded script. I couldn't figure out how to make it work, so I'd say we should let Eelco handle this and possibly tell us how to use it for the future.

**Comment by edolstra** [comment]
Here is the update to the fallback paths:
```
commit 40174f8ab475e97baabfee0283afd07489db9534 (HEAD -> master)
Author: Eelco Dolstra <edolstra@gmail.com>
Date:   Tue Apr 19 23:48:50 2022 +0200

    nix-fallback-paths.nix: Update to 2.8.0

diff --git a/nixos/modules/installer/tools/nix-fallback-paths.nix b/nixos/modules/installer/tools/nix-fallback-paths.nix
index dfafda77cb5..1707935ad5b 100644
--- a/nixos/modules/installer/tools/nix-fallback-paths.nix
+++ b/nixos/modules/installer/tools/nix-fallback-paths.nix
@@ -1,7 +1,7 @@
 {
-  x86_64-linux = "/nix/store/0n2wfvi1i3fg97cjc54wslvk0804y0sn-nix-2.7.0";
-  i686-linux = "/nix/store/4p27c1k9z99pli6x8cxfph20yfyzn9nh-nix-2.7.0";
-  aarch64-linux = "/nix/store/r9yr8ijsb0gi9r7y92y3yzyld59yp0kj-nix-2.7.0";
-  x86_64-darwin = "/nix/store/hyfj5imsd0c4amlcjpf8l6w4q2draaj3-nix-2.7.0";
-  aarch64-darwin = "/nix/store/9l96qllhbb6xrsjaai76dn74ap7rq92n-nix-2.7.0";
+  x86_64-linux = "/nix/store/yx36yzxpw1hn4fz8iyf1rfyd56jg3yf4-nix-2.8.0";
+  i686-linux = "/nix/store/c0hg806zvwg800qbszzj8ff4a224kjgf-nix-2.8.0";
+  aarch64-linux = "/nix/store/wic2832ll53q392r2wks4xr2nrk7p8p5-nix-2.8.0";
+  x86_64-darwin = "/nix/store/5yqdvnkmkrhl36xh0qy31pymdphjimdd-nix-2.8.0";
+  aarch64-darwin = "/nix/store/izc9592szrnpv8n86hr88bhpyc9g6b4s-nix-2.8.0";
 }

```

**Comment by edolstra** [comment]
We should remove Nix 2.4 to 2.7. Our policy is not to keep old versions around in Nixpkgs.

**Comment by edolstra** [comment]
And while we're at it, we can remove `nix-unstable` since the frequent release schedule makes it unnecessary. `nix-unstable` is often older than the most recent stable release...

**Comment by drupol** [comment]
> We should remove Nix 2.4 to 2.7. Our policy is not to keep old versions around in Nixpkgs.

I guess you meant from 2.3 to 2.7 ?

Could you please review what I did in the PR and let me know if it's ok ?

Thanks!

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -38,59 +38,10 @@ let
         aws-sdk-cpp = aws-sdk-cpp-nix;
       };
 in lib.makeExtensible (self: {
-  nix_2_3 = (common rec {
```
Comment (lovesegfault) [inline pkgs/tools/package-management/nix/default.nix:41]: NAK.

Please don't do this.

Just bump the package and leave the rest of this alone.

It has been discussed multiple times and many people rely on those.

**Review by lovesegfault — CHANGES_REQUESTED** [review]
NAK on deleting other variants.

**Comment by lovesegfault** [comment]
Let's move the removal to another PR, we can discuss it there. We shouldn't bundle them with a simple version bump anyway.

File: pkgs/tools/package-management/nix/default.nix (hunk 1 above)
Comment (drupol) [reply pkgs/tools/package-management/nix/default.nix:41]: So, I restore from 2.3 to 2.7 ?

**Review by lovesegfault — APPROVED** [review]

**Comment by drupol** [comment]
Ok, going to squash my commits.

**Pull request merged**